catalog/dags/maintenance/update_science_museum_urls.py

```diff
@@ -0,0 +1,164 @@
+
+Once complete, we can use the `invalid_science_museum_ids` to identify records to delete. They are not automatically deleted by this DAG, in order to give us an opportunity to first see how many there are.
+"""
+
+import logging
+from datetime import timedelta
+from textwrap import dedent
+
+from airflow.decorators import dag, task
+from airflow.exceptions import AirflowSkipException
+from airflow.models.abstractoperator import AbstractOperator
+from airflow.models.param import Param
+
+from common import slack
+from common.constants import DAG_DEFAULT_ARGS, POSTGRES_CONN_ID
+from common.sql import RETURN_ROW_COUNT, PostgresHook
+from common.urls import rewrite_redirected_url
+
+
+logger = logging.getLogger(__name__)
+
+
+DAG_ID = "update_science_museum_urls"
+INVALID_IDS_TABLE = "science_museum_invalid_ids"
+
+UPDATE_URLS_QUERY = dedent(
+    """
+    UPDATE image
+    SET url=REPLACE(url, '/images', '')
+    WHERE provider='sciencemuseum' AND url ILIKE '%/images/%';
+    """
+)
+CREATE_TABLE_QUERY = dedent(
+    f"""
+    CREATE TABLE IF NOT EXISTS {INVALID_IDS_TABLE}
+    (identifier uuid);
+    """
+)
+
+
+@task
+def run_sql(
+    sql: str,
+    handler: callable = None,
+    postgres_conn_id: str = POSTGRES_CONN_ID,
+    task: AbstractOperator = None,
+):
+    postgres = PostgresHook(
+        postgres_conn_id=postgres_conn_id,
+        default_statement_timeout=PostgresHook.get_execution_timeout(task),
+    )
+
+    return postgres.run(sql, handler=handler)
+
+
+@task
+def get_batches(
+    batch_size: int,
+    postgres_conn_id: str = POSTGRES_CONN_ID,
+    task: AbstractOperator = None,
+):
+    postgres = PostgresHook(
+        postgres_conn_id=postgres_conn_id,
+        default_statement_timeout=PostgresHook.get_execution_timeout(task),
+    )
+    records = postgres.get_records(
+        "SELECT identifier, url FROM image where provider='sciencemuseum';"
+    )
+
+    return [records[i : i + batch_size] for i in range(0, len(records), batch_size)]
+
+
+@task
+def process_batch(records):
+    invalid_ids = []
+    for identifier, url in records:
+        # Failed to reach URL
+        if rewrite_redirected_url(url) is None:
+            invalid_ids.append(identifier)
+
+    return invalid_ids
+
+
+@task
+def record_invalid_ids(invalid_ids):
+    # Chain together
+    ids_to_record = sum(invalid_ids, [])
+
+    if not ids_to_record:
+        raise AirflowSkipException("No invalid urls found!")
+
+    values = ", ".join([f"('{id}')" for id in ids_to_record])
+    query = dedent(
+        f"""
+        INSERT INTO {INVALID_IDS_TABLE} (identifier)
+        VALUES {values}
+        """
+    )
+
+    return run_sql.function(sql=query, handler=RETURN_ROW_COUNT)
+
+
+@task
+def notify_slack(invalid_count: int):
+    slack.send_message(
+        f"Detected {invalid_count} invalid Science Museum urls.", dag_id=DAG_ID
+    )
+
+
+@dag(
+    dag_id=DAG_ID,
+    schedule=None,
+    catchup=False,
+    tags=["maintenance"],
```

Comment:
`maintenance` feels like an odd tasks for a data consistency issue, given the other DAGs we've tagged with it... 🤔 The `add_license_url` DAG has the `data_normalization` tag, maybe we should use that? Or `data_consistency`?

Reply:
I actually changed it to `maintenance` from `data_normalization` because I wasn't sure if it was wrong to associate it with that specific project 😆 But I think you're right, that's probably a bitter fit than `maintenance`. Works for me, will update.